### PR TITLE
Make tl.debug_barrier() a no-op on CPU

### PR DESCRIPTION
**Pull request by int3**
Arguably there's a case for implementing it on CPU (e.g. to debug cache coherency issues) but I don't think it's worthwhile at the moment. It's also being used as a workaround for https://github.com/triton-lang/triton/issues/1615 but it's unclear to me if that is a CUDA-specific issue (the reproducer in the issue doesn't seem to repro any more).

I plan to upstream this to make future merges easier, but I would like some feedback first on whether my approach makes sense.

### CODE REVIEW

File: python/triton/language/semantic.py [hunk 1]
```diff
@@ -1519,7 +1519,9 @@ def max_constancy(x: tl.tensor, values: List[int]) -> tl.tensor:
 
 
 def debug_barrier(builder: ir.builder) -> tl.tensor:
-    return tl.tensor(builder.create_barrier(), tl.void)
+    if builder.options.supports_barrier:
+        return tl.tensor(builder.create_barrier(), tl.void)
+    return tl.tensor(builder.get_null_value(builder.get_int32_ty()), tl.int32)
```
Comment (int3) [inline python/triton/language/semantic.py:1524]: not sure if there's a better way to create a no-op

**Comment by ienkovich** [comment]
If you feel there is a potential use for debug barriers for CPU (and actually we might want to have multi-threaded kernels in the future), then we should probably drop `BarrierOp` somewhere in our pipeline and leave front-end as is

File: third_party/cpu/lib/TritonCPUToLLVM/DebugOpsToLLVM.cpp [hunk 2]
```diff
@@ -164,6 +166,23 @@ struct PrintOpConversion : public ConvertOpToLLVMPattern<triton::PrintOp> {
   }
 };
 
+using BarrierOp = mlir::gpu::BarrierOp;
```
Comment (int3) [inline third_party/cpu/lib/TritonCPUToLLVM/DebugOpsToLLVM.cpp:169]: there are two namespaces called `gpu` (a top-level one, and `mlir::gpu`), so we have to use the full namespace path here

**Comment by int3** [comment]
Fair enough; I've updated it to drop it in the backend.

**Review by ienkovich — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**